app/Platform/Commands/SyncTriggers.php

```diff
@@ -0,0 +1,195 @@
+                    $newTrigger = (new UpsertTriggerVersionAction())->execute(
+                        $leaderboard,
+                        $leaderboard->Mem,
+                        versioned: true,
+                    );
+
+                    // Try our best to backdate the new trigger's timestamps.
+                    if ($newTrigger) {
+                        // First determine the creation date.
+                        $createdTimestamp =
+                            $leaderboard->Created
+                            ?? $leaderboard->Updated
+                            ?? now();
+
+                        // Then ensure updated is never before created.
+                        $updatedTimestamp = max(
+                            $createdTimestamp,
+                            $leaderboard->comments->first()?->Submitted
+                            ?? $leaderboard->Updated
+                            ?? $createdTimestamp
+                        );
+
+                        $newTrigger->timestamps = false;
+                        $newTrigger->update([
+                            'created_at' => $createdTimestamp,
+                            'updated_at' => $updatedTimestamp,
+                        ]);
+                    }
+
+                    $progressBar->advance();
+                }
+            });
+
+        $progressBar->finish();
+        $this->newLine();
+        $this->info('Done syncing triggers for leaderboards.');
+    }
+
+    private function syncRichPresenceTriggers(): void
+    {
+       $gamesCount = Game::whereNotNull('RichPresencePatch')
+           ->where('RichPresencePatch', '!=', '')
+           ->count();
+
+       $this->info("Syncing triggers for {$gamesCount} rich presence scripts...");
+
+       $progressBar = $this->output->createProgressBar($gamesCount);
+
+       Game::query()
+           ->whereNotNull('RichPresencePatch')
+           ->where('RichPresencePatch', '!=', '')
+           ->with(['modificationsComments' => function ($query) {
+                $query->automated()
+                    ->whereRaw("LOWER(Payload) LIKE '%changed the rich presence%'")
+                    ->latest('Submitted')
+                    ->limit(1);
+            }])
+           ->chunk(1000, function ($games) use ($progressBar) {
+               foreach ($games as $game) {
+                   $newTrigger = (new UpsertTriggerVersionAction())->execute(
+                       $game,
+                       $game->RichPresencePatch,
+                       versioned: true
+                   );
+
+                    // Try our best to backdate the new trigger's timestamps.
+                    if ($newTrigger) {
+                        // When RP is edited, a comment is left. Use that comment's submitted
+                        // date, otherwise we just have to fall back to the current date.
+                        $timestamp = $game->modificationsComments->first()?->Submitted ?? now();
```

Comment:
I wonder if these should fall back to `$game->created_at` instead of `now()`? Looks like quite a few games have null creation times, and a game may sit around for a while before someone works on it.

Maybe the created timestamp of the oldest achievement in the set?

Reply:
Updated in latest to:
```php
// When RP is edited, a comment is left. Use that comment's submitted
// date, otherwise fall back to various dates of increasingly lesser precision.
$timestamp =
    $game->modificationsComments->first()?->Submitted
    ?? $game->achievements()->min('DateCreated')
    ?? $game->Created
    ?? $game->Updated;
```